akka-typed/src/main/scala/akka/typed/cluster/sharding/ClusterSharding.scala

```diff
@@ -0,0 +1,323 @@
+   * @param typeName A name that uniquely identifies the type of entity in this cluster
+   * @param handOffStopMessage Message sent to an entity to tell it to stop
+   * @tparam A The type of command the entity accepts
+   */
+  // TODO: FYI, I think it would be very good to have rule that "behavior, otherstuff" 
+  // TODO: or "behavior, props, otherstuff" be the consistent style we want to promote in parameter ordering, WDYT?
+  def spawn[A](
+    behavior:           Behavior[A],
+    props:              Props,
+    typeName:           String,
+    settings:           ClusterShardingSettings,
+    maxNumberOfShards:  Int,
+    handOffStopMessage: A): ActorRef[ShardingEnvelope[A]]
+
+  /**
+   * Spawn a shard region or a proxy depending on if the settings require role and if this node has such a role.
+   *
+   * @param behavior The behavior for entities
+   * @param typeName A name that uniquely identifies the type of entity in this cluster
+   * @param entityProps Props to apply when starting an entity
+   * @param allocationStrategy Allocation strategy which decides on which nodes to allocate new shards
+   * @param handOffStopMessage Message sent to an entity to tell it to stop
+   * @tparam E A possible envelope around the message the entity accepts
+   * @tparam A The type of command the entity accepts
+   */
+  def spawn[E, A](
+    behavior:           Behavior[A],
+    entityProps:        Props,
+    typeName:           String,
+    settings:           ClusterShardingSettings,
+    messageExtractor:   ShardingMessageExtractor[E, A],
+    allocationStrategy: ShardAllocationStrategy,
+    handOffStopMessage: A
+  ): ActorRef[E]
+
+  /**
+   * Spawn a shard region or a proxy depending on if the settings require role and if this node has such a role.
+   *
+   * @param behavior The behavior for entities
+   * @param typeName A name that uniquely identifies the type of entity in this cluster
+   * @param entityProps Props to apply when starting an entity
+   * @param handOffStopMessage Message sent to an entity to tell it to stop
+   * @tparam E A possible envelope around the message the entity accepts
+   * @tparam A The type of command the entity accepts
+   */
+  def spawn[E, A](
+    behavior:           Behavior[A],
+    entityProps:        Props,
+    typeName:           String,
+    settings:           ClusterShardingSettings,
+    messageExtractor:   ShardingMessageExtractor[E, A],
+    handOffStopMessage: A
+  ): ActorRef[E]
+
+  /**
+   * Create an `ActorRef`-like reference to a specific sharded entity.
+   * Messages sent to it will be wrapped in a [[ShardingEnvelope]] and passed to the local shard region or proxy.
+   *
+   * Note: FIXME explain why it's not the same as actor ref
+   *
+   * This way of addressing Sharded Entities
+   */
+  def entityRefFor[A](typeName: String, entityId: String): ShardedEntityRef[A]
+  // FIXME this is not so nice to use from Java, a class param would be welcome there
```

Comment:
this raises the question if we do scaladsl/javadsl ClusterSharding or add a method here;
Method here probably is fine, with get prefix.